packages/delegation-process/src/services/readModelService.ts

```diff
@@ -573,6 +574,29 @@ export function readModelServiceBuilder(
         ),
       };
     },
+    async getDelegationRelatedAgreement(
+      eserviceId: EServiceId,
+      consumerId: TenantId
+    ): Promise<Agreement | null> {
+      const data = await agreements.findOne({
+        "data.eserviceId": eserviceId,
+        "data.consumerId": consumerId,
+        "data.state": agreementState.active,
```

Comment:
According [to the SRS](https://pagopa.atlassian.net/wiki/spaces/PDNDI/pages/1236959344/SRS+Incaricato#Un-fruitore-delega-un-altro-ente-alla-fruizione-di-un-e-service), we should prevent the creation of a delegation if there exists an agreement in state Active, Suspended or Pending